### PR TITLE
Abductor Gear now works on borers

### PR DESCRIPTION
**Pull request by Super3222**
Requires https://github.com/yogstation13/yogstation/pull/1148 to be merged, which is entirely my fault. If 1448 doesn't go through I'll make a new PR for this change.

### Intent of your Pull Request

- Stun Mode on Advanced Baton stuns and weakens borers.
- Sleep Mode on Advanced Baton mutes borers.
- Probe Mode can detect borers.

#### Changelog

:cl:
tweak: Advanced Batons stun and sleep modes work on borers.
tweak: Stamina Loss in general now mutes borers.
tweak: Advanced Batons probe mode can detect borers inside of bodies.
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
This is stupid. Sleeping a human doesn't sleep a xeno larva in them, and it certainly doesn't directly hit the brain with enough force to hurt the borer.

**Comment by Super3222** [comment]
Xeno Larvas aren't even on the same playing field as borers. Xeno Larva's also aren't even sentient until they are born, so making them "sleep" while they are growing would be redundant.

As for this change, it only makes sense for abductors to have the technology and capability to counter borers. On a hierarchy of aliens in the SS13 universe, abductors rank far above almost everything in the game. They are known for knowing about everything, aside from occasional plasma facilities which they are only beginning to study and research.

Going away from a lore sense, it's perfect for game sense. Because borers shouldn't have the capability of jumping out of their host, freezing an abductor, and than jumping into their body without a fair fight.

And may I remind you that advanced batons _are highly advanced pieces of alien technology suitable for taking down any human subject and kidnapping them_. Holy shit, even a medical HUD can catch a borer. You're telling me abductor gear can't go a step further?

Lastly, never did I officially make borers completely immobile to the threat of abductors. Stun mode only _stuns them._ Sleep mode only _mutes them._ Nothing is too extreme

Edit:
In the last commit.
- Borers will now become mute from stamina loss. Given that they'll barely ever get stamina loss, and getting stamina loss won't ever have a different side effect, it now makes them mute. This is an alternative to my other method of making them mute, which wasn't working since simple mobs don't have a `silent` variable.

**Comment by ShadowDeath6** [comment]
just make it to where actually hitting the borer with the baton stuns it, it can't do anything while in a sleeping host anyway and abductors have ample time to stun the borer if they're both present

**Comment by Kmc2000** [comment]
You induce sleep in penis.

Induce, as in this implies a neurological function or at least a nervous signal, as borers are connected to the host's nervous and neurological systems it makes logical sense that they are affected.

They crawl in your ear and attach to your brain from christ's sakes

**Comment by ShadowDeath6** [comment]
borers aren't connected enough to the brain to actually feel what the host feels though, else ODing on synaptizine would kill the borer

**Comment by Kmc2000** [comment]
Nerdy response out of the way, it makes sense in the context of the game

**Comment by ShadowDeath6** [comment]
fine, I doubt this will be a big problem anyway considering abductors hardly ever probe their targets

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 1]
```diff
@@ -455,6 +460,10 @@ Congratulations! You are now trained for xenobiology research!"}
 							"<span class='userdanger'>You suddenly feel very drowsy!</span>")
 		playsound(loc, 'sound/weapons/Egloves.ogg', 50, 1, -1)
 		L.Sleeping(sleeptime)
+		if(iscarbon(L))
+			var/mob/living/carbon/C = L
+			if(C.borer)
+				C.borer.staminaloss += sleeptime
```
Comment (Cruix) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:466]: Simple animals don't reduce their staminaloss in every Life() tick, only carbons do, so this will mute them forever.

File: code/game/gamemodes/miniantags/abduction/abduction_gear.dm [hunk 2]
```diff
@@ -437,6 +437,11 @@ Congratulations! You are now trained for xenobiology research!"}
 	L.Stun(stuntime)
 	L.Weaken(stuntime)
 	L.apply_effect(STUTTER, stuntime)
+	if(iscarbon(L))
+		var/mob/living/carbon/C = L
+		if(C.borer)
+			C.borer.Stun(stuntime)
+			C.borer.Weaken(stuntime)
```
Comment (ShadowDeath6) [inline code/game/gamemodes/miniantags/abduction/abduction_gear.dm:444]: I was under the impression that simplemobs can't be stunned or weakened

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 3]
```diff
@@ -107,6 +107,13 @@
 			handle_automated_movement()
 			handle_automated_action()
 			handle_automated_speech()
+
+		if(staminaloss)
+			adjustStaminaLoss(-3)
+		if(weakened)
+			AdjustWeakened(-1)
+		if(stunned)
+			AdjustStunned(-1)
```
Comment (Cruix) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: stunned and weakened are handled in /mob/living/Life()

Comment (Cruix) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: stunned and weakened are handled in /mob/living/Life(). You should probably move the staminaloss thing to handle_status_effects() instead of putting it in Life().
